Add support for medaka.

dodoradio opened the pull request:
This adds LCD code for medaka, the WSD-F21HR, successor to koi/ayu. 
The time sync code is identical to koi.
The display colour code uses a different set of bits to encode black vs. white. 
The 'timepiece mode' command is still 'coming soon'. It is quite different, as it needs to select whether it wants to show the date/barometer and battery/step/altimeter data on the top and bottom displays.

The watch isn't actually supported by asteroid yet, but I'm working on cleaning up the port and hopefully that will be added soon.

dodoradio committed:
Add support for medaka.
This adds LCD code for medaka, the WSD-F21HR, successor to koi/ayu.
The time sync code is identical to koi.
The display colour code uses a different set of bits to encode black vs. white.
The 'timepiece mode' command is still 'coming soon'. It is quite different, as it needs to select whether it wants to show the date/barometer and battery/step/altimeter data on the top and bottom displays.

Signed-off-by: Arseniy Movshev <dodoradio@outlook.com>

MagneFire commented on src/medaka-tools.cpp:
```diff
@@ -0,0 +1,52 @@
+#include <QDate>
+#include <QFile>
+
+#include<chrono>
```
Inconsistent spacing.

MagneFire replied:
Maybe even remove these imports since they do not appear to be used?

MagneFire commented on src/medaka-tools.cpp:
```diff
@@ -0,0 +1,52 @@
+	(uint8_t) ((dateNow.month() << 5) | (dateNow.day() & 0x1F)),
+	(uint8_t) (((dateNow.dayOfWeek() % 7) << 5) | (timeNow.hour() & 0x1F)), //weeks are counted from sunday, sunday encoded as zero.
+	(uint8_t) (timeNow.minute() & 0xFF),
+	(uint8_t) ((timeNow.second()) & 0xFF)
```
This line appears to be the only one that's different compared to `koi`. Was this on purpose?

dodoradio replied:
yes. medaka's display updates faster, so the 1s offset isn't relevant

MagneFire commented on src/medaka-tools.cpp:
```diff
@@ -0,0 +1,52 @@
+	multiSensorFile.close();
+}
+
+static void KoiSet12H(bool value) {
```
Can you rename this to `Medaka`

dodoradio replied:
oops

MagneFire reviewed (commented):
Minor nitpicks.

The code between `koi` and `medaka` look very similar (`Write` and `SyncTime`).
It makes me wonder if it would make sense to just rename the `koi-tools` to `casio-tools`. Use the same namespace in the argument parser. But then provide different functions where necessary. Leaving this up to you and other to decide.

dodoradio commented:
only write is shared. The timesync command is intentionally different. It might make sense to unify those, but I'm not sure.
there will be a lot more differences coming soon. I already wrote (but didn't PR) a set of additional functions which allowed me to enable additional functions such as the compass mode or the arbitrary string display: On the surface, it seems that quite a few of these features are shared, if not completely identical. However, the architecture of how these commands are interpreted is different. For example, on koi, setting the display colour acts immediately. on medaka, it will only act after the colour display turns on and off. also, koi's display immediately returns to timekeeping mode when it leaves the additional display modes, but medaka's needs to be explicitly reinitialised to timekeeping mode. Because of these differences I chose to separate the two watches, as a sort of 'don't assume these work the same way!', but I understand this looks a bit strange without explanation. 
I can, however, try to unify the two in some capacity, but I'm not sure it will actually make the code cleaner or easier to maintain, it will just get rid of duplication for the sake of getting rid of duplication.

dodoradio committed:
koi: remove unnecessary dlfcn import
This seems to have been left over from when this code was merged with catfish, but wasn't ever used

dodoradio commented:
@MagneFire I've now cleaned up the code significantly. I've also added a separate commit which removes an unneeded dlfcn import from koi. 

I want to keep the koi and medaka code entirely separate, at the cost of a few duplicated lines.
 Koi and Medaka's displays actually operate a bit more differently under the hood than it may seem from this commit, so I'd like to separate them now to avoid having to untangle them from each other if/when I add a few more functions (I've already got some of this additional functionality written, but I'm not sure if this repo is the best place to put that kind of thing).
Personally I think it's also a bit easier to understand this code when each watch's implementation is entirely done within a single file, keeping cross-references to a minimum, but that's a personal thing and I'm open to merging some of this code.
